shell/server/proxy.js

```diff
@@ -936,13 +937,69 @@ Proxy.prototype._callNewWebSession = function (request, userInfo) {
                                 WebSession.typeId, params).session;
 };
 
+var isRfc1918OrLocal = function(address) {
+  if (Net.isIPv4(address)) {
+    quad = address.split(".").map(function(x) { return parseInt(x, 10); });
+    return (quad[0] === 127 || quad[0] === 10 ||
+            (quad[0] === 192 && quad[1] === 168) ||
+            (quad[0] === 172 && quad[1] >= 16 && quad[1] < 32));
+  } else if (Net.isIPv6(address)) {
+    // IPv6 specifies ::1 as localhost and fd:: as reserved for private networks
+    return (address === "::1" || address.indexOf("fd") === 0);
```

Comment:
Ah, searching backwards starting at 0 (and ending at 0) means testing precisely what we want.

Cool!  Today I learned.